guava/src/com/google/common/collect/Iterators.java

```diff
@@ -1303,4 +1304,38 @@ public ConcatenatedIterator(Iterator<? extends Iterator<? extends T>> iterators)
   static <T> ListIterator<T> cast(Iterator<T> iterator) {
     return (ListIterator<T>) iterator;
   }
+  
+  
+  /**
+   * Converts an Integer iterator into a corresponding primitive iterator
+   * @param iterator iterator to convert
+   * @return converted iterator
+   */
+  public static PrimitiveIterator.OfInt toIntIterator(Iterator<Integer> iterator) {
+	  checkNotNull(iterator);
+	  List<Integer> list = Lists.newArrayList(iterator);
+	  return list.stream().mapToInt(n-> n).iterator();
+  }
+  
+  /**
+   * Converts a Long iterator into a corresponding primitive iterator
+   * @param iterator iterator to convert
+   * @return converted iterator
+   */
+  public static PrimitiveIterator.OfLong toLongIterator(Iterator<Long> iterator) {
+	  checkNotNull(iterator);
+	  List<Long> list = Lists.newArrayList(iterator);
```

Comment:
Why need collect iterator to list? May be better create stream from Iterable.spliterator() :
```java
        Iterable<Long> longs = () -> iterator;
        return StreamSupport.stream(longs.spliterator(), false)
                .mapToLong(n -> n)
                .iterator();
```